packages/nextjs/hooks/scaffold-eth/useTransactor.tsx

```diff
@@ -0,0 +1,108 @@
+import { TransactionRequest, TransactionResponse, TransactionReceipt } from "@ethersproject/abstract-provider";
+import { SendTransactionResult } from "@wagmi/core";
+import { BigNumber, ethers, Signer } from "ethers";
+import { Deferrable } from "ethers/lib/utils";
+import { useSigner } from "wagmi";
+import toast from "react-hot-toast";
+import { getParsedEthersError } from "~~/components/scaffold-eth/Contract/utilsContract";
+import { toast as customToast } from "~~/components/scaffold-eth/index";
+
+const DEBUG = true;
+
+type TTransactionFunc = (
+  tx: Promise<SendTransactionResult> | Deferrable<TransactionRequest> | undefined,
+  callback?: ((_param: any) => void) | undefined,
+) => Promise<Record<string, any> | undefined>;
+
+export const useTransactor = (_signer?: Signer, gasPrice?: number): TTransactionFunc | undefined => {
+  /**
+   * ### Summary 
+    if signer is provided in params use it(this means dev wants to send a raw transaction)
+    else use the signer from connect wallet (mostly its an wagmi writeTxn)
+   */
+  let signer = _signer;
+  const { data } = useSigner();
+  if (signer === undefined && data) {
+    signer = data;
+  }
+
+  const result: TTransactionFunc = async (tx, callback) => {
+    let toastId = null;
+    let transactionReceipt: TransactionReceipt | undefined;
+    let transactionResponse: SendTransactionResult | TransactionResponse | undefined;
+    if (signer) {
+      try {
+        const chainId = await signer.getChainId();
+        const provider = signer.provider;
+        const networkName = (await provider?.getNetwork())?.name;
+
+        let etherscanNetwork = "";
+        if (networkName && chainId && chainId > 1) {
+          etherscanNetwork = networkName + ".";
+        }
+
+        let etherscanTxUrl = "https://" + etherscanNetwork + "etherscan.io/tx/";
+        if (chainId && chainId === 100) {
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          etherscanTxUrl = "https://blockscout.com/poa/xdai/tx/";
```

Comment:
I think we can abstract this into a util. (`networks.ts`?)

It can have a `getBlockExplorerTxLink` method